okhttp-logging-interceptor/src/main/java/com/squareup/okhttp/logging/HttpLoggingInterceptor.java

```diff
@@ -199,9 +200,12 @@ public void setLevel(Level level) {
 
         Charset charset = UTF8;
         MediaType contentType = responseBody.contentType();
-        if (contentType != null) {
-          charset = contentType.charset(UTF8);
-        }
+        if (contentType != null)
+          try {
+            charset = contentType.charset(UTF8);
+          } catch (UnsupportedCharsetException e) {
+            // unsupported charset, assume UTF8 nevertheless
```

Comment:
Should we log that the media type is probably broken? I'd love to just log something like “couldn't decode the response body; charset is likely malformed”.

Otherwise developers aren't motivated to fix their broken servers.